package/MDAnalysis/coordinates/RDKit.py

```diff
@@ -95,10 +132,353 @@ def __init__(self, filename, **kwargs):
+        value = value.item()
+    if attr == "names":
+        # RDKit needs the name to be properly formated for a
+        # PDB file (1 letter elements start at col 14)
+        name = re.findall(r'(\D+|\d+)', value)
+        if len(name) == 2:
+            symbol, number = name
+        else:
+            symbol, number = name[0], ""
+        value = "{:>2.2}{:<2.2}".format(symbol, number)
+    # set attribute value in RDKit MonomerInfo
+    rdattr = RDATTRIBUTES[attr]
+    getattr(mi, "Set%s" % rdattr)(value)
+
+
+def _set_atom_property(atom, attr, value):
+    """Converts an MDAnalysis atom attribute into an RDKit atom property"""
+    if isinstance(value, (float, np.float)):
+        atom.SetDoubleProp("_MDAnalysis_%s" % attr, float(value))
+    elif isinstance(value, (int, np.int)):
+        atom.SetIntProp("_MDAnalysis_%s" % attr, int(value))
+    else:
+        atom.SetProp("_MDAnalysis_%s" % attr, value)
+
+
+def _infer_bo_and_charges(mol, terminal_atom_indices=[]):
+    """Infer bond orders and formal charges from a molecule.
+
+    Since most MD topology files don't explicitely retain informations on bond
+    orders or charges, it has to be guessed from the topology. This is done by
+    looping other each atom and comparing its expected valence to the current
+    valence to get the Number of Unpaired Electrons (NUE).
+    If an atom has a negative NUE, it needs a positive formal charge (-NUE).
+    If two neighbouring atoms have UEs, the bond between them most
+    likely has to be increased by the value of the smallest NUE.
+    If after this process, an atom still has UEs, it's either a radical
+    (because one its bonds was cut when creating the AtomGroup) or it needs a
+    negative formal charge of -NUE. Since these radical atoms can be detected
+    when looping over the bonds of the AtomGroup, only atoms that are not part
+    of this "terminal_atoms" list will be assigned a negative formal charge.
+
+    Parameters
+    ----------
+
+    mol : rdkit.Chem.rdchem.RWMol
+        The molecule is modified inplace and must have all hydrogens added
+
+    terminal_atom_indices : list
+        List of terminal atoms indices, i.e. atoms at the edges of a molecule
+    """
+
+    for atom in mol.GetAtoms():
+        # get NUE for each possible valence
+        expected_vs = PERIODIC_TABLE.GetValenceList(atom.GetAtomicNum())
+        current_v = atom.GetTotalValence()
+        nue = [v - current_v for v in expected_vs]
+
+        # if there's only one possible valence state and the corresponding
+        # NUE is negative, it means we can only add a positive charge to
+        # the atom
+        if (len(nue) == 1) and (nue[0] < 0):
+            atom.SetFormalCharge(-nue[0])
+            mol.UpdatePropertyCache(strict=False)
+            continue
+        else:
+            neighbors = atom.GetNeighbors()
+            # check if one of the neighbors has a common NUE
+            for i, na in enumerate(neighbors, start=1):
```

Comment:
It mostly makes sense to me, but I'm having trouble wrapping my head around whether or not this could lead to order specific behavoiur i.e. raising the bond order for one neighbour pair means that another won't be raised. I can't really think of a case where this would be a problem though.

Do you have any thoughts here?

Reply:
It is indeed order-specific, for example for a carboxylate group `R-C(-O)-O` the first oxygen read will receive a double bond and the other one will be charged. Of course IRL both oxygens share the charge and the bond order but that's not something RDKit was made to handle. In other words, I don't think there's anything I can do about it...

Reply:
Can we put this down in the docstring as a Note? If anything it might be useful to know for some user at some point.